pkg/usrlocalsharelima/usrlocalsharelima.go

```diff
@@ -23,32 +23,42 @@ import (
 // It will also append the file extension on Windows, if necessary.
 // This function is different from os.Executable(), which will use /proc/self/exe on Linux
 // and therefore will resolve any symlink used to locate the executable. This function will
-// return the symlink instead because we want to locate ../share/lima relative to the location
-// of the symlink, and not the actual executable. This is important when using Homebrew.
-//
-// If os.Args[0] is invalid, this function still falls back on os.Executable().
+// return the symlink instead because we want to be able to locate ../share/lima relative
+// to the location of the symlink, and not the actual executable. This is important when
+// using Homebrew.
 var executableViaArgs0 = sync.OnceValues(func() (string, error) {
 	if os.Args[0] == "" {
-		logrus.Warn("os.Args[0] has not been set")
-	} else {
-		executable, err := exec.LookPath(os.Args[0])
-		if err == nil {
-			// LookPath() will add the `.exe` file extension on Windows, but will not return an
-			// absolute path if the argument contained any of `:/\` (or just `/` on Unix).
-			return filepath.Abs(executable)
-		}
-		logrus.Warnf("os.Args[0] is invalid: %v", err)
+		return "", errors.New("os.Args[0] has not been set")
+	}
+	executable, err := exec.LookPath(os.Args[0])
+	if err == nil {
+		executable, err = filepath.Abs(executable)
+	}
+	if err != nil {
+		return "", fmt.Errorf("os.Args[0] is invalid: %w", err)
 	}
-	return os.Executable()
+
+	return executable, nil
 })
 
 // Dir returns the location of the <PREFIX>/lima/share directory, relative to the location
 // of the current executable. It checks for multiple possible filesystem layouts and returns
 // the first candidate that contains the native guest agent binary.
 func Dir() (string, error) {
-	self, err := executableViaArgs0()
+	selfPaths := []string{}
+
+	selfViaArgs0, err := executableViaArgs0()
 	if err != nil {
-		return "", err
+		logrus.WithError(err).Warnf("failed to find executable from os.Args[0]: %v", err)
```

Comment:
`err` is logged twice:
```suggestion
		logrus.WithError(err).Warn("failed to find executable from os.Args[0]")
```